### ISSUE TITLE
Unify tooling

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#1640]
We already import the heavy golangci-lint in our tooling:

https://github.com/open-telemetry/opentelemetry-go/blob/992837f195eacbd84f0596bfaa79470dba7408e5/internal/tools/tools.go#L21

We also import directly misspell, which is a linter [bundled with golangci-lint](https://golangci-lint.run/usage/linters/). Additionally, [it has been proposed](https://github.com/open-telemetry/opentelemetry-go/pull/1489/files) to included staticcheck, another linter included with golangci-lint.

Instead we should update our golangci-lint configuration to included these and other useful linters and remove the direct dependency on misspell.

**Comment by lastchiliarch** [comment]
> > What linters are we going to enable except staticcheck? I'd like to help if possible. @MrAlias
> 
> @lastchiliarch can you audit what is available and make a recommendation here? I would like to start with a minimal set that covers the other checks we are currently doing and anything else that seems like a gap.

There are many linters, the full list is here: https://golangci-lint.run/usage/linters/.

I think the linters which were enabled by default is very necessary, https://golangci-lint.run/usage/linters/#enabled-by-default-linters.
lll,  ifshort, gomnd, gocyclo, whitespace are also in my recommend list.

Belove  are something  'problems' that there linters find :

```

internal/global/trace.go:142:2: variable 'delegate' is only used in the if-statement (internal/global/trace.go:143:2); consider using short syntax (ifshort)
	delegate := t.delegate.Load()

attribute/set.go:357:7: mnd: Magic number: 2, in <case> detected (gomnd)
	case 2:
	     ^
attribute/set.go:381:7: mnd: Magic number: 8, in <case> detected (gomnd)
	case 8:
	     ^
attribute/set.go:389:7: mnd: Magic number: 10, in <case> detected (gomnd)
	case 10:
	     ^


propagation/trace_context.go:46: line is 156 characters (lll)
var traceCtxRegExp = regexp.MustCompile("^(?P<version>[0-9a-f]{2})-(?P<traceID>[a-f0-9]{32})-(?P<spanID>[a-f0-9]{16})-(?P<traceFlags>[a-f0-9]{2})(?:-.*)?$")
propagation/baggage.go:72:23: mnd: Magic number: 2, in <condition> detected (gomnd)
		if len(nameValue) < 2 {
		                    ^
propagation/trace_context.go:93:20: mnd: Magic number: 5, in <condition> detected (gomnd)
	if len(matches) < 5 { // four subgroups plus the overall match

```

**Comment by MrAlias** [comment]
Closed with #2598. The bundled misspell is not equivalent to the one installed and the operation cannot be performed on markdown files.
